jjrunner.py

```diff
@@ -0,0 +1,198 @@
+#    run scripts
+
+builtin_vars = [
+    "BRANCH_NAME", "CHANGE_ID", "CHANGE_URL", "CHANGE_TITLE", "CHANGE_AUTHOR",
+    "CHANGE_AUTHOR_DISPLAY_NAME", "CHANGE_AUTHOR_EMAIL", "CHANGE_TARGET",
+    "BUILD_NUMBER", "BUILD_ID", "BUILD_DISPLAY_NAME", "JOB_NAME",
+    "JOB_BASE_NAME", "BUILD_TAG", "EXECUTOR_NUMBER", "NODE_NAME",
+    "NODE_LABELS", "WORKSPACE", "JENKINS_HOME", "JENKINS_URL", "BUILD_URL",
+    "JOB_URL", "GIT_COMMIT", "GIT_PREVIOUS_COMMIT",
+    "GIT_PREVIOUS_SUCCESSFUL_COMMIT", "GIT_BRANCH", "GIT_LOCAL_BRANCH",
+    "GIT_URL", "GIT_COMMITTER_NAME", "GIT_AUTHOR_NAME", "GIT_COMMITTER_EMAIL",
+    "GIT_AUTHOR_EMAIL"]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Execute CI jobs locally')
+    parser.add_argument('jobname',
+                        help='name of the job that you want to run locally')
+    parser.add_argument('--args', '-a',
+                        help='JSON dictionary to override job arguments')
+    parser.add_argument('--dryrun', '-d', action='store_true',
+                        help=('dryrun with all command and parameters '
+                              'saved in /tmp directory'))
+    parser.add_argument('--reason', '-r', help=('override build reason'),
+                        default=None)
+    args = parser.parse_args()
+
+    jobname = args.jobname
+    print("ARGS: jobname: [%s], args [%s]" % (args.jobname, args.args))
+
+    src = 'https://ci.openquake.org/'
+    jjr_user = os.getenv('JJR_USER', None)
+    jjr_pass = os.getenv('JJR_PASS', None)
+    if jjr_user is None or jjr_pass is None:
+        print("JJR_USER and/or JJR_PASS environment variables are not set")
+        sys.exit(1)
+    server = Jenkins(src, username=jjr_user, password=jjr_pass)
+    job_conf = server.get_job_config(jobname)
+    tree = ElementTree.fromstring(job_conf)
+
+    configs = tree.find('properties').find(
+        'hudson.model.ParametersDefinitionProperty').find(
+            'parameterDefinitions')
+
+    # collect params from job config
+    params = OrderedDict()
+
+    params['GEM_JENKINS_REASON'] = {
+        'name': 'GEM_JENKINS_REASON',
+        'desc': 'auto-generated'}
+    if args.reason is None:
+        params['GEM_JENKINS_REASON']['defa'] = (
+            'Started by user %s' % getpass.getuser())
+    else:
+        params['GEM_JENKINS_REASON']['defa'] = args.reason
+
+    params['JOB_NAME'] = {'name': 'JOB_NAME', 'defa': jobname,
+                          'desc': 'auto-generated'}
+    params['JENKINS_HOME'] = {'name': 'JENKINS_HOME', 'defa': expanduser("~"),
+                              'desc': 'auto-generated'}
+    params['BUILD_NUMBER'] = {'name': 'BUILD_NUMBER', 'defa': 1,
+                              'desc': 'auto-generated'}
+    try:
+        g = git.cmd.Git(os.getcwd())
+        params['GIT_BRANCH'] = {'name': 'GIT_BRANCH',
+                                'defa': g.rev_parse('--abbrev-ref', 'HEAD'),
+                                'desc': 'auto-generated'}
+    except:
```

Comment:
Wouldn't be better to trap specific exceptions?

Reply:
retrieve GIT_BRANCH value is optional; so, if something goes wrong, the fallback is always print the WARNING and go ahead.